automata/fa/nfa.py

```diff
@@ -216,21 +217,165 @@ def from_regex(
         Self
             The NFA accepting the language of the input regex.
         """
+        # Import the shorthand character classes
+        from automata.regex.parser import (
+            DIGIT_CHARS,
+            NON_DIGIT_CHARS,
+            NON_WHITESPACE_CHARS,
+            NON_WORD_CHARS,
+            WHITESPACE_CHARS,
+            WORD_CHARS,
+        )
+
+        # Create a set for additional symbols from shorthand classes
+        additional_symbols = set()
+
+        # Check for shorthand classes in the regex
+        if "\\s" in regex:
+            additional_symbols.update(WHITESPACE_CHARS)
+        if "\\S" in regex:
+            additional_symbols.update(NON_WHITESPACE_CHARS)
+        if "\\d" in regex:
+            additional_symbols.update(DIGIT_CHARS)
+        if "\\D" in regex:
+            additional_symbols.update(NON_DIGIT_CHARS)
+        if "\\w" in regex:
+            additional_symbols.update(WORD_CHARS)
+        if "\\W" in regex:
+            additional_symbols.update(NON_WORD_CHARS)
```

Comment:
@StefanosChaliasos Can you please refactor this to use a `dict`-based lookup table? That would make this much less repetitive.

cc @eliotwrobson

Reply:
I agree this could make things much cleaner, especially since this can be done in a loop 👍🏽